lib/config.rb

```diff
@@ -48,6 +49,14 @@ def self.load_files(*sources)
 
     config.add_source!(Sources::EnvSource.new(ENV)) if Config.use_env
 
+    if defined?(::Rails::Railtie) && Config.use_rails_credentials
+      if Rails.version < '7.1'
+        config.add_source!(Sources::HashSource.new(secret: Rails.application.secrets.to_h.deep_stringify_keys))
```

Comment:
Let's remove the nesting of the credentials under the top-level `secret` key. I think that'll be more useful as the default behavior in order to support merging with other config sources. For example, if my config.yaml file has

```yaml
aws:
  region: 'us-west-1'
```

And my Rails credentials has

```yaml
aws:
  secret_access_key: '123456'
```

...then I'd want to be able to access `Settings.aws`, which would contain both `region` and `secret_access_key`.

If someone _really_ wants to keep their secrets in a separate key, they can always turn off `use_rails_credentials` add it themselves in an initializer

```ruby
Settings.add_source!({ secret: Rails.application.credentials.config.deep_stringify_keys })
Settings.reload!
```

Reply:
Awesome, agree on this.